Concentration/ViewController.swift

```diff
@@ -9,58 +9,104 @@ import UIKit
 
 class ViewController: UIViewController {
     
-    lazy var game = Concentration(numberOfPairsOfCards: (cardButtons.count + 1) / 2)
+    var gameTimer: Timer?
     
-    var flipCount = 0 {
-        didSet{
-            flipCountLabel.text = "Flips: \(flipCount)"
-        }
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        startNewGame()
+        gameTimer = Timer.scheduledTimer(timeInterval: 5, target: self, selector: #selector(updateScoreForTime), userInfo: nil, repeats: true)
+    }
+    
+    override func viewWillDisappear(_ animated: Bool) {
+        super.viewWillDisappear(animated)
+        gameTimer?.invalidate()
```

Comment:
I think that we want to start a new timer each time the user starts a new game. Am I wrong?

Reply:
@alonreik now the timer is restarted for every game?

Reply:
yes